metricq_source_ipmi/source/main.py

```diff
@@ -381,6 +436,8 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.period = None
         self.result_queue = Queue()
+        self.collection_loops: Optional[asyncio.Task] = None
```

Comment:
Why is it called a `loop` even though the type clearly states that it is a task?

Reply:
Why is the name in plural, but the type indicate only one object?

Reply:
> Why is the name in plural, but the type indicate only one object?

`self.collection_loops` contains the task that waits for all of the collection loops to finish:

https://github.com/metricq/metricq-source-ipmi/pull/15/files/b15333f94bda9f869ac758963eac9b2b34444510#diff-708aa81b7fbba1b95f6a477c68d89d78R262-R266

So yes, it is a single `Task`, but logically it is just the set of all log loops running concurrently, thus `collection_loops` (plural).

We could have `self.collection_loops: Set[asyncio.Task]` and spawn/cancel them one by one so that type and name match up, but that feels like reinventing `asyncio.gather`.